platform/saivpp/vpplib/SwitchStateBaseFdb.cpp

```diff
@@ -1082,3 +1110,264 @@ sai_status_t SwitchStateBase::vpp_delete_bvi_interface(
 
     return SAI_STATUS_SUCCESS;
 }
+
+uint32_t SwitchStateBase::lag_to_bond_if_idx (const sai_object_id_t lag_id)
+{
+    auto it = m_lag_bond_map.find(lag_id);
+
+    if (it == m_lag_bond_map.end())
+    {
+        SWSS_LOG_ERROR("failed to find bond if idx for lag id: %x",lag_id);
+	return ~0;
+    }
+    return it->second;
+}
+
+int SwitchStateBase::remove_lag_to_bond_entry(const sai_object_id_t lag_oid)
+{
+   auto it = m_lag_bond_map.find(lag_oid);
+
+    if (it == m_lag_bond_map.end())
+    {
+        SWSS_LOG_ERROR("failed to find lag swif index for : %s", sai_serialize_object_id(lag_oid).c_str());
+        return ~0;
+    }
+
+    SWSS_LOG_NOTICE("Removing lag object swif index: %s", sai_serialize_object_id(lag_oid).c_str());
+    m_lag_bond_map.erase(it);
+    return 0;
+}
+
+sai_status_t SwitchStateBase:: createLag(
+        _In_ sai_object_id_t object_id,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list)
+{
+    SWSS_LOG_ENTER();
+
+    auto sid = sai_serialize_object_id(object_id);
+    CHECK_STATUS(create_internal(SAI_OBJECT_TYPE_LAG, sid, switch_id, attr_count, attr_list));
+    return vpp_create_lag(object_id, attr_count, attr_list);
+
+}
+
+sai_status_t SwitchStateBase::vpp_create_lag(
+	_In_ sai_object_id_t lag_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list)
+{
+    uint32_t bond_id, mode, lb;
+    uint32_t swif_idx = ~0;
+    const char *hw_ifname;
+    SWSS_LOG_ENTER();
+
+    //set mode and lb. ONiC config does not have provision to pass mode and load balancing algorithm
+    mode = VPP_BOND_API_MODE_ROUND_ROBIN;
+
+    //if LACP is to be enabled in VPP
+    //mode = VPP_BOND_API_MODE_LACP;
+    lb = VPP_BOND_API_LB_ALGO_L2;
```

Comment:
I think the hash algorithm should cover L4 and L4 headers for flow consistency to avoid reordering problem.

Reply:
Ideally this should be configurable , which SONiC does not support. I am keeping round-robin as default one. Based on the requirement one has to make it configurable which SONiC does not support. May be we will raise an issue to track it.

Reply:
There is no one-size-fit-all solution here. If the LAG is for L2 switching, we should use L23. If it is for L3 routing, L34 is better, which is the typical 5 tuples hashing. But we don't know if a LAG is for L2 or L3 at the time it is created. Maybe we can create lag with a default lb algo like l23 and change algo to l34 if the lag has IP configured? To do that, either we have to delete and recreate lag in vpp or we augment vpp to allow changing lb algo after lag is created.